com.unity.render-pipelines.high-definition/Editor/Material/Decal/DecalProjectorEditor.cs

```diff
@@ -1,15 +1,62 @@
 using UnityEngine;
-using UnityEngine.Rendering;
 using UnityEngine.Rendering.HighDefinition;
 using UnityEditor.ShortcutManagement;
 using static UnityEditorInternal.EditMode;
+using UnityEditor.IMGUI.Controls;
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Linq.Expressions;
 
 namespace UnityEditor.Rendering.HighDefinition
 {
     [CustomEditor(typeof(DecalProjector))]
     [CanEditMultipleObjects]
     partial class DecalProjectorEditor : Editor
     {
+        const float k_Limit = 100000;
+        const float k_LimitInv = 1 / k_Limit;
+
+        static object s_ColorPref;
+        static Func<Color> GetColorPref;
+        static Color fullColor
+        {
+            get
+            {
+                Color c = s_LastColor;
+                c.a = 1;
+                return c;
+            }
+        }
+        static Color s_LastColor;
+        static void UpdateColorsInHandlesIfRequired()
+        {
+            Color c = GetColorPref();
+            if (c != s_LastColor)
+            {
+                if (s_BoxHandle != null && !s_BoxHandle.Equals(null))
+                    s_BoxHandle = null;
+
+                if (s_UVHandles != null && !s_UVHandles.Equals(null))
+                    s_UVHandles.baseColor = c;
+
+                s_LastColor = c;
+            }
+        }
+
+        static DecalProjectorEditor()
+        {
+            // PrefColor is the type to use to have a Color that is customizable inside the Preference/Colors panel.
+            // Sadly it is internal so we must create it and grab color from it by reflection.
+            Type prefColorType = typeof(Editor).Assembly.GetType("UnityEditor.PrefColor");
+            s_ColorPref = Activator.CreateInstance(prefColorType, new object[] { "HDRP/Decal", k_GizmoColorBase.r, k_GizmoColorBase.g, k_GizmoColorBase.b, k_GizmoColorBase.a });
```

Comment:
I think @MaddalenaUnity wants to unify the colors under `Rendering`, could you double check?